graph/TypeGraph.java

```diff
@@ -406,8 +406,8 @@ public References references() {
 
         public FunctionalIterator<RuleStructure> all() {
             Encoding.Prefix index = IndexIID.Rule.prefix();
-            FunctionalIterator<RuleStructure> persistedRules = storage.iterate(index.bytes(), (key, value) ->
-                    convert(StructureIID.Rule.of(value)));
+            FunctionalIterator<RuleStructure> persistedRules = storage.iterate(index.bytes(), Storage.SortedPair::new)
```

Comment:
we don't want necessarily make `StructureIID` implements `ByteComparable`, so have a sortable pair that we can then use a normal `map` operation over it, discarding the sortedness property we don't care about in this case